### ISSUE TITLE
Storefront single product pagination next previous sorting inverted

### ISSUE THREAD

**Issue by ntindicator** [woocommerce/storefront#1765]
## Describe the bug
When viewing the single product page, the previous and next product pagination links are pointing in the opposite directions to the sorting on the shop loop page. This seems confusing for the user because the product to the left is the previous product (next in loop), and the product to the right is the next (previous in loop). It makes more sense to have the next on the left, and previous on the right. This means that the user is scrolling back through the catalog by clicking the right product link. This is mainly a problem is sorting options are disabled, and products are always sorted by date. In this case the single product pagination could follow the sorting if referred from a loop page, as detailed in this [issue](https://github.com/woocommerce/storefront/issues/983).

Isolating the problem (mark completed items with an [x]):
- [x ] I have deactivated other plugins and themes and confirmed this bug occurs when only WooCommerce + Storefront theme are active.
- [ x] I can reproduce this bug consistently using the steps below.

## To Reproduce
1. Create products with unique timestamps (if the timestamps aren't unique consistent sorting isn't possible)
2. Load Shop page
3. Change default sorting to sort by latest
4. Select a product link that isn't the first or most recent
5. Note the products to the left and right of this product
6. View single product page of selected product
7. Check the direction of left and right pagination links

## Expected behavior
Expected single product pagination to match direction of latest sorting on shop page

### Browser Environment
Please provide as much detail as possible about your testing environment.

- Platform: macOS
- Browser(s): Firefox

### WordPress Environment
Please provide relevant details of your WordPress setup and server environment.

<details>
```
Copy and paste the system status report from WooCommerce > System Status in WordPress admin.
`
### WordPress Environment ###

WordPress address (URL): http://one.wordpress.test
Site address (URL): http://one.wordpress.test
WC Version: 5.6.0
REST API Version: ✔ 5.6.0
WC Blocks Version: ✔ 5.7.0
Action Scheduler Version: ✔ 3.2.1
WC Admin Version: ✔ 2.5.1
Log Directory Writable: ✔
WP Version: 5.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.21.1
PHP Version: 7.4.22
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.68.0
OpenSSL/1.1.1f

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.5.12-MariaDB-1:10.5.12+maria~focal
Max Upload Size: 1 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.6.0
WC Database Prefix: wp_
Total Database Size: 6.05MB
Database Data Size: 3.75MB
Database Index Size: 2.30MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.03MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.81MB + Engine InnoDB
wp_posts: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.16MB + Index: 0.11MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 484

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (1) ###

WooCommerce: by Automattic – 5.7.0-dev

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 4.1.10
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /?page_id=6
Cart: #7 - /?page_id=7
Checkout: #8 - /?page_id=8
My account: #9 - /?page_id=9
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.8.0 (update to version 3.8.1 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 68
Oldest: 2021-08-19 10:12:04 +0000
Newest: 2021-08-21 08:27:55 +0000


### Status report information ###

Generated at: 2021-08-21 08:54:03 +00:00
`
```
</details>

**Comment by ralucaStan** [comment]
Hi there @ntindicator,
Thank you for reaching out and taking the time to create this issue.

The pagination on the single product view is intended to be a navigation in the product's category. The previous/next products are the ones that have a date before/after the date of the current product. So the sorting on the shop page should not influence it.

There is a known [issue regarding](https://github.com/woocommerce/storefront/issues/1375) regarding this pagination when the products are imported so check that out for further details.

As this is not a bug, I am going to go and close this ticket.

**Comment by ntindicator** [comment]
Navigating to the right on the latest post always takes the user to a post from a long time ago (the first post in the category). Clicking the link to the right should be to the previous post if they’re sorted from latest to first, in my opinion. I agree the loop sorting is not connected, but in my case it is, as I removed the sorting options, and use date sorting only. Also I’d prefer to remove the latest product linking to the first, and have no link in that direction for that post. I’m also aware of this issue you mentioned, and only discovered the inverted links when I started looking into that issue.
